xonsh/tools.py

```diff
@@ -1560,6 +1560,17 @@ def __init__(self):
         self._alias_checksum = None
         self._path_mtime = -1
 
+    @staticmethod
+    def get_possible_names(key):
+        if ON_WINDOWS:
+            key = key.upper()
+            return {
+                key + ext
+                for ext in ([''] + builtins.__xonsh_env__['PATHEXT'])
+            }
```

Comment:
Oh, you mean this ordering actually conveys a preference! Makes sense, will change.